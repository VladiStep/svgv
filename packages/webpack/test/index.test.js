const path = require('path');
const { compile, CompileError } = require('./utils.js');

const loader = path.resolve(__dirname, '../src/index.js');

describe('loader', () => {
  it('should transform svg', async () => {
    const result = await compile('raw.svg', ['vue-loader', loader]);
    expect(result.join('\n\n---\n\n')).toMatchSnapshot('result');
  }, 15000);

  it('should allow options', async () => {
    let result = await compile('raw.svg', [
      'vue-loader',
      {
        loader,
        options: {
          svgo: {
            plugins: ['removeViewBox'],
          },
        },
      },
    ]);
    expect(result.join('\n\n---\n\n')).toMatchSnapshot('removeViewBox');

    result = await compile('raw.svg', [
      'vue-loader',
      {
        loader,
        options: {
          svgo: {
            plugins: null,
          },
        },
      },
    ]);
    expect(result.join('\n\n---\n\n')).toMatchSnapshot('null');
  }, 15000);

  it('should support url-loader', async () => {
    const result = await compile('raw.svg', [
      'vue-loader',
      loader,
      'url-loader',
    ]);
    expect(result.join('\n\n---\n\n')).toMatchSnapshot('result');
  }, 15000);

  it('should support file-loader', async () => {
    const result = await compile('raw.svg', [
      'vue-loader',
      loader,
      'file-loader',
    ]);
    expect(result.join('\n\n---\n\n')).toMatchSnapshot('result');
  }, 15000);

  it('should throw compile error', async () => {
    await expect(
      compile('invalid.svg', ['vue-loader', loader])
    ).rejects.toThrow(CompileError);
  }, 15000);
});
