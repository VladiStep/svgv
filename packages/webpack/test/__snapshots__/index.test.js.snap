// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should allow options: null 1`] = `
"import { render } from \\"./raw.svg?vue&type=template&id=56df30f5\\"
const script = {}
script.render = render

export default script

---

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

const _hoisted_1 = {
  xmlns: \\"http://www.w3.org/2000/svg\\",
  viewBox: \\"0 0 100 100\\",
  width: \\"100px\\",
  height: \\"100px\\"
}
const _hoisted_2 = /*#__PURE__*/_createVNode(\\"circle\\", {
  cx: \\"50\\",
  cy: \\"50\\",
  r: \\"50\\",
  style: {\\"fill\\":\\"#00f\\"}
}, null, -1 /* HOISTED */)

export function render(_ctx, _cache) {
  return (_openBlock(), _createBlock(\\"svg\\", _hoisted_1, [
    _hoisted_2
  ]))
}"
`;

exports[`loader should allow options: removeViewBox 1`] = `
"import { render } from \\"./raw.svg?vue&type=template&id=8ffa4cac\\"
const script = {}
script.render = render

export default script

---

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

const _hoisted_1 = {
  xmlns: \\"http://www.w3.org/2000/svg\\",
  width: \\"100px\\",
  height: \\"100px\\"
}
const _hoisted_2 = /*#__PURE__*/_createVNode(\\"circle\\", {
  cx: \\"50\\",
  cy: \\"50\\",
  r: \\"50\\",
  style: {\\"fill\\":\\"#00f\\"}
}, null, -1 /* HOISTED */)

export function render(_ctx, _cache) {
  return (_openBlock(), _createBlock(\\"svg\\", _hoisted_1, [
    _hoisted_2
  ]))
}"
`;

exports[`loader should support file-loader: result 1`] = `
"import { render } from \\"./raw.svg?vue&type=template&id=b5f1ca76\\"
import script from \\"./raw.svg?vue&type=script&lang=js\\"
export * from \\"./raw.svg?vue&type=script&lang=js\\"
script.render = render

export default script

---

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

const _hoisted_1 = {
  xmlns: \\"http://www.w3.org/2000/svg\\",
  viewBox: \\"0 0 100 100\\",
  width: \\"100\\",
  height: \\"100\\"
}
const _hoisted_2 = /*#__PURE__*/_createVNode(\\"circle\\", {
  cx: \\"50\\",
  cy: \\"50\\",
  r: \\"50\\",
  fill: \\"#00f\\"
}, null, -1 /* HOISTED */)

export function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (_openBlock(), _createBlock(\\"svg\\", _hoisted_1, [
    _hoisted_2
  ]))
}

---

export default new String(__webpack_public_path__ + \\"261ed99e5d6e1bdf643add2822dbbf55.svg\\");"
`;

exports[`loader should support url-loader: result 1`] = `
"import { render } from \\"./raw.svg?vue&type=template&id=3bf96b22\\"
import script from \\"./raw.svg?vue&type=script&lang=js\\"
export * from \\"./raw.svg?vue&type=script&lang=js\\"
script.render = render

export default script

---

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

const _hoisted_1 = {
  xmlns: \\"http://www.w3.org/2000/svg\\",
  viewBox: \\"0 0 100 100\\",
  width: \\"100\\",
  height: \\"100\\"
}
const _hoisted_2 = /*#__PURE__*/_createVNode(\\"circle\\", {
  cx: \\"50\\",
  cy: \\"50\\",
  r: \\"50\\",
  fill: \\"#00f\\"
}, null, -1 /* HOISTED */)

export function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (_openBlock(), _createBlock(\\"svg\\", _hoisted_1, [
    _hoisted_2
  ]))
}

---

export default new String(\\"data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPCFET0NUWVBFIHN2ZyBQVUJMSUMgIi0vL1czQy8vRFREIFNWRyAxLjEvL0VOIiAiaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkIj4KPHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDAgMTAwIiB3aWR0aD0iMTAwcHgiIGhlaWdodD0iMTAwcHgiPgogIDxjaXJjbGUgY2xhc3M9ImNpcmNsZSIgY3g9IjUwIiBjeT0iNTAiIHI9IjUwIiAvPgogIDxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI+CiAgLmNpcmNsZSB7CiAgICBmaWxsOiBibHVlOwogIH0KICAudW5rbm93biB7CiAgICBmaWxsOiByZWQ7CiAgfQogIDwvc3R5bGU+CiAgPHNjcmlwdD5jb25zb2xlLmVycm9yKCdzdmcnKTs8L3NjcmlwdD4KPC9zdmc+Cg==\\");"
`;

exports[`loader should transform svg: result 1`] = `
"import { render } from \\"./raw.svg?vue&type=template&id=ceb116a4\\"
const script = {}
script.render = render

export default script

---

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

const _hoisted_1 = {
  xmlns: \\"http://www.w3.org/2000/svg\\",
  viewBox: \\"0 0 100 100\\",
  width: \\"100\\",
  height: \\"100\\"
}
const _hoisted_2 = /*#__PURE__*/_createVNode(\\"circle\\", {
  cx: \\"50\\",
  cy: \\"50\\",
  r: \\"50\\",
  fill: \\"#00f\\"
}, null, -1 /* HOISTED */)

export function render(_ctx, _cache) {
  return (_openBlock(), _createBlock(\\"svg\\", _hoisted_1, [
    _hoisted_2
  ]))
}"
`;
